server/datastore/mysql/apple_mdm.go

```diff
@@ -2078,8 +2078,6 @@ func (ds *Datastore) bulkSetPendingMDMAppleHostProfilesDB(
 		hmap.host_uuid IN (?) %s AND
 		-- profiles that are in B but not in A
 		ds.profile_uuid IS NULL AND ds.host_uuid IS NULL AND
-		-- except "remove" operations in any state
-		( hmap.operation_type IS NULL OR hmap.operation_type != ? ) AND
```

Comment:
I think this combination should be unique, but in a convoluted way? That is, we have this:

```
CREATE TABLE `host_mdm_apple_profiles` (
...
PRIMARY KEY (`host_uuid`,`profile_uuid`),
```

So a host <-> profile combination is guaranteed unique, and then in `mdm_apple_configuration_profiles` we have this:

```
UNIQUE KEY `idx_mdm_apple_config_prof_team_identifier` (`team_id`,`identifier`),
```

So a profile `identifier` is unique for a given team, and a host is part of a single team. It is technically still not guaranteed, I guess, because the host could have been transferred and have a `host_uuid + profile identifier` of another team installed, and the same profile identifier exists in its new team. But if there is a clash, it is necessarily in a "to be removed" vs "to be installed" profiles, which may not cause issues in the intersection processing? Bit of a brain teaser tbh, not sure.

Reply:
That being said, not clear to me why we remove this condition?

Reply:
> That being said, not clear to me why we remove this condition?

This condition assumes that we'll be preemptively bulk deleting by profile identifier, which is where we've see the bottleneck. If we want to be more selective/targeted with our deletes, we
need to use the bimap to be able detect cases where we have the same profile identifier but different profile uuid and so this query needs to include removes because it is how we populate the bimap.

Reply:
Gotcha, thanks!